src/rest.rs

```diff
@@ -936,6 +936,80 @@ fn handle_request(
             json_response(prepare_txs(txs, query, config), TTL_SHORT)
         }
 
+        (&Method::GET, Some(script_types @ &"addresses"), Some(&"txs"), None, None, None)
+        | (&Method::GET, Some(script_types @ &"scripthashes"), Some(&"txs"), None, None, None) => {
+            let script_type = match *script_types {
+                "addresses" => "address",
+                "scripthashes" => "scripthash",
+                _ => "",
+            };
+            let script_hashes: Vec<[u8; 32]> = query_params
+                .get(*script_types)
```

Comment:
Maybe we should limit the number of addresses/scripthashes?

Realistically, how many addresses does a caller need to include in the query?

If we want to make this configurable per-instance, it needs to go in config.

Also, if we expect a ton of addresses... maybe it should be a POST request? I think the limit on GET parameter total size is kinda small...

Reply:
updated to use POST instead of GET

Reply:
agree we should limit the number of addresses, although I'm not sure what would be a sensible limit.

the main use case for this PR needs to fetch transactions for ~200 addresses, and it would simplify my life considerably not to have to deduplicate and splice back together results of multiple paginated queries, so hopefully at least that many :P

also imo it might be better hard-coded rather than configurable, since the client needs to know in advance.